v3/newrelic/sqlparse/sqlparse.go

```diff
@@ -55,15 +176,32 @@ var (
 // Ability to correctly parse queries for other SQL databases is not
 // guaranteed.
 func ParseQuery(segment *newrelic.DatastoreSegment, query string) {
-	s := cCommentRegex.ReplaceAllString(query, "")
-	s = lineCommentRegex.ReplaceAllString(s, "")
-	s = sqlPrefixRegex.ReplaceAllString(s, "")
-	op := strings.ToLower(firstWordRegex.FindString(s))
-	if rg, ok := sqlOperations[op]; ok {
+	sql := sqlTokenizer{query: query}
+	op := strings.ToLower(removeAllComments(sql.nextWord()))
```

Comment:
Why are we calling `removeAllComments` when `nextWord` already skips leading and trailing comments? Won't this just result in unnecessary scans of the word?

Reply:
You're right, that's unnecessary. I removed the call.